### ISSUE TITLE
twisted.web.xmlrpc.withRequest takes extra arguments and has no @since tag

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4855]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4855|
|Type|defect|
|Created|2011-02-08 14:26:25Z|

`withRequest` is a decorator, but it takes `*args, **kwargs` for some reason.  The docstring also doesn't note that it was added in 10.2.

Attachments:

* [withrequest-since.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/456/45604c57f22a3d83d904bea7fe0cd2fdec33bfab/f82a957ac9640025ada7b208d66060864e29458c.patch) (723 bytes) - added by exarkun on 2011-03-01 16:39:58Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4855 4855
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1297175185000000 1297175185000000
changetime__1299984456000000 1299984456000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez set owner to @moshez<br>@moshez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)<a name="note_6"></a>|@moshez set owner to @exarkun<br>@moshez set status to `new`|
|-|-|

Looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [30971]) Remove extraneous arguments from withRequest signature and mark the version in which it was introduced. 

Author: exarkun
Reviewer: moshez
Fixes: [#4855](https://github.com/twisted/twisted/issues/4855)
